core/src/main/java/org/axonframework/common/lock/PessimisticLockFactory.java

```diff
@@ -29,46 +31,50 @@
 
 /**
  * Implementation of a {@link LockFactory} that uses a pessimistic locking strategy. Calls to
- * {@link #obtainLock} will block until a lock could be obtained. If a lock is obtained by a thread, that
- * thread has guaranteed unique access.
+ * {@link #obtainLock} will block until a lock could be obtained or back off based on the {@link BackoffParameters}
+ * by throwing an exception. The later will cause the command to fail, but will allow the Thread to be released. If a
+ * lock is obtained by a thread, that thread has guaranteed unique access.
  * <p/>
  * Each thread can hold the same lock multiple times. The lock will only be released for other threads when the lock
  * has been released as many times as it was obtained.
  * <p/>
  * This lock can be used to ensure thread safe access to a number of objects, such as Aggregates and Sagas.
  *
+ * Back off properties with respect acquiring locks can be configured though the {@link BackoffParameters}.
+ *
  * @author Allard Buijze
  * @since 1.3
  */
 public class PessimisticLockFactory implements LockFactory {
 
-    private static final Set<PessimisticLockFactory> INSTANCES =
-            newSetFromMap(synchronizedMap(new WeakHashMap<>()));
+    private static final Set<PessimisticLockFactory> INSTANCES = newSetFromMap(synchronizedMap(new WeakHashMap<>()));
 
-    private static Set<Thread> threadsWaitingForMyLocks(Thread owner) {
-        return threadsWaitingForMyLocks(owner, INSTANCES);
-    }
+    private final ConcurrentHashMap<String, DisposableLock> locks = new ConcurrentHashMap<>();
+    private final BackoffParameters backoffParameters;
 
-    private static Set<Thread> threadsWaitingForMyLocks(Thread owner, Set<PessimisticLockFactory> locksInUse) {
-        Set<Thread> waitingThreads = new HashSet<>();
-        for (PessimisticLockFactory lock : locksInUse) {
-            lock.locks.values().stream()
-                    .filter(disposableLock -> disposableLock.isHeldBy(owner))
-                    .forEach(disposableLock -> disposableLock.queuedThreads().stream()
-                            .filter(waitingThreads::add)
-                            .forEach(thread -> waitingThreads.addAll(threadsWaitingForMyLocks(thread, locksInUse))));
-        }
-        return waitingThreads;
+    /**
+     * Creates a new IdentifierBasedLock instance.
+     * <p/>
+     * Deadlocks are detected across instances of the IdentifierBasedLock.
+     * This constructor specifies no back off from lock acquisition
+     *
+     * @apiNote here for backwards compatibility
```

Comment:
I would argue that the default should be some back off (no matter how high the upper-bound would be set) If things go haywire your only option becomes restarting the application as it will not just never recover; it will converge to a state where the application won't process any command at all anymore.

Reply:
Yes, I do agree that eventually a default which does not run indefinite is valuable, but that did not follow from this bit of javadoc alone on face value. 
I am just not sure whether _this_ form of documentation or this specific description will provide the required feedback that on a future axon 4.0 release we will make that adjustment.

I'd say that making the intent of that note a bit more clear justifies keeping it in place.
As it as right now, I think I stay by my initial point.